### DISCUSSION TITLE
How do i add a panel to a particular group?

### DISCUSSION THREAD

**Discussion by badmonster0** [mathuo/dockview#635] [General]
[export const ResourcePanel = ({ ...props }) => {
    const panelParams = props.params;

    const [panelId, setPanelId] = useState(panelParams.panelId);
    return (
        <PanelContext.Provider value={{ panelId, setPanelId }}>
            <Resource nodeId={panelParams.resourceId} viewSize={"L"} redirectMode="live" />
        </PanelContext.Provider>
    );
}
](https://dockview.dev/docs/core/panels/add#relative-to-another-panel)

I tried this and it always adds it to a new group, but not the same group as the reference panel.

**Comment by mathuo** [comment]
Do you the full example of the code you are trying? The approaches to adding panels to particular groups described [here](https://dockview.dev/docs/core/panels/add/#relative-to-another-panel) should work just fine

**Comment by badmonster0** [comment]
[sandbox link](https://codesandbox.io/p/sandbox/simple-dockview-forked-h745ps?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cly5dsgu500063b6jv4tjlsur%2522%252C%2522sizes%2522%253A%255B100%252C0%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cly5dsgu400023b6jlz9t31ce%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cly5dsgu400033b6jfuq5u0qp%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cly5dsgu400053b6jo0zlhtdk%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cly5dsgu400023b6jlz9t31ce%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cly5dsgu400013b6jpewniy10%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522cly5dsgu400023b6jlz9t31ce%2522%252C%2522activeTabId%2522%253A%2522cly5dsgu400013b6jpewniy10%2522%257D%252C%2522cly5dsgu400053b6jo0zlhtdk%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cly5dsgu400043b6jiq4vbkhl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522cly5dsgu400053b6jo0zlhtdk%2522%252C%2522activeTabId%2522%253A%2522cly5dsgu400043b6jiq4vbkhl%2522%257D%252C%2522cly5dsgu400033b6jfuq5u0qp%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cly5dsgu400033b6jfuq5u0qp%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

Here is an example, how do i add "Panel 4" to the same group as "Panel 1" explicitly?

I tried
```
   event.api.addPanel({
      id: "panel_4",
      component: "default",
      params: { color: "red" },
      position: { referencePanel: "panel_1", direction: "right" }
    });
   ```
    
    and 

```
   event.api.addPanel({
      id: "panel_4",
      component: "default",
      params: { color: "red" },
      position: { referenceGroup: "panel_1", direction: "right" }
    });
    ```
    as the document, both of them add a new group and position that to panel1's group.

**Reply by badmonster0** [reply]
Should i use "within"? (btw where do i find the documentation for meanings of all directions?)

**Comment by pele1410** [comment]
I am also interested in this behavior.  Is there any update?

**Comment by mathuo** [comment]
Specifying the `direction: "within"` or not including a `direction` field should both result in the new panel being added with the referenceGroup or referencePanel
